plasmapy/simulation/particle_tracker.py

```diff
@@ -169,6 +170,75 @@ def total(self) -> float:
         return float(self.tracker.nparticles)
 
 
+class AllParticlesOffGridTerminationCondition(AbstractTerminationCondition):
```

Comment:
There's kind of a lot of these already, and I expect we may add more in the future! To head off a potential future refactor, what if we moved these files into different modules now: 

simulation/particle_tracker/particle_tracker.py 
simulation/particle_tracker/termination_conditions.py 
simulation/particle_tracker/save_routines.py

Reply:
How should we handle the namespace? Should all of this still be accessible with something like
```py
from plasmapy.simulation.particle_tracker import AbstractSaveRoutine
```
or were you thinking
```py
from plasmapy.simulation.particle_tracker.save_routines import AbstractSaveRoutine
```

Reply:
I think the latter? Separate documentation pages would be good so it's not too cluttered on the main particle_tracker page.